packages/swapper/src/swappers/zrx/getSendMaxAmount/getSendMaxAmount.ts

```diff
@@ -0,0 +1,62 @@
+import { chainAdapters, ChainTypes, SendMaxAmountInput } from '@shapeshiftoss/types'
+import BigNumber from 'bignumber.js'
+
+import { SwapError } from '../../../api'
+import { ZrxSwapperDeps } from '../ZrxSwapper'
+
+export async function getSendMaxAmount(
+  { adapterManager }: ZrxSwapperDeps,
+  {
+    wallet,
+    quote,
+    sellAssetAccountId,
+    feeEstimateKey = chainAdapters.FeeDataKey.Average
+  }: SendMaxAmountInput
+): Promise<string> {
+  const adapter = adapterManager.byChain(ChainTypes.Ethereum)
+  const bip32Params = adapter.buildBIP32Params({ accountNumber: Number(sellAssetAccountId) })
+  const ethAddress = await adapter.getAddress({ wallet, bip32Params })
+
+  const account = await adapter.getAccount(ethAddress)
+  const tokenId = quote.sellAsset?.tokenId
+
+  let balance: string | undefined
+  if (tokenId) {
+    balance = account.chainSpecific.tokens?.find(
+      (token) => token.contract.toLowerCase() === tokenId?.toLowerCase()
+    )?.balance
+  } else {
+    balance = account.balance
+  }
+
+  if (!balance) {
+    throw new SwapError(`No balance found for ${tokenId ? quote.sellAsset.symbol : 'ETH'}`)
+  }
+
+  // return the erc20 token balance. No need to subtract the fee.
+  if (tokenId && new BigNumber(balance).gt(0)) {
+    return balance
+  }
+
+  let feeEstimates
+  try {
+    feeEstimates = await adapter.getFeeData({
+      to: quote.depositAddress,
+      from: ethAddress,
+      value: balance,
+      contractAddress: tokenId
+    })
+  } catch (err) {
+    throw new SwapError(`getSendMaxAmount:getFeeData - ${err}`)
+  }
+
+  const estimatedFee = feeEstimates[feeEstimateKey].chainSpecific.feePerTx
+
+  const sendMaxAmount = new BigNumber(balance).minus(estimatedFee)
+
+  if (!sendMaxAmount.gt(0)) {
+    throw new SwapError('ETH balance is less than estimated fee')
+  }
+
+  return sendMaxAmount.toString()
```

Comment:
Why not return the BigNumber?

Reply:
We currently return strings for all our numerical values. So this is staying with that pattern.